### ISSUE TITLE
Uyuni: login less often

### ISSUE THREAD

**Issue by roidelapluie** [prometheus/prometheus#9540]
The uyuni SD currently logs in and out on every call.

Ideally, we would use a custom roundtripper that would log in and keep the received token as long as the token is valid.

**Comment by David-N-Perkins** [comment]
I'm new to Prometheus and thought this might be a good place to get started. I found the `config.NewRoundTripperFromConfig` method that creates custom roundtrippers for different auth approaches, but uyuni SD uses an RPC client with the auth token included as the first argument in every call. I'm not very familiar with RPC, but I didn't see a way to put this in the roundtripper, because the token is put in the request body with all the other arguments. So, I'm assuming this would need to be implemented in uyuni SD.

I looked at their [API](https://www.uyuni-project.org/uyuni-docs-api/uyuni/api/auth.html) and theirs very little information about the returned token or error responses if the token is expired. So I was thinking of adding a member variable to keep track of how old the token is and refreshing when it approached a hard coded limit. How does that sound?

I'm also not sure how to test this. The existing unit tests are pretty sparse and spinning up a Uyuni server looks pretty involved. Normally, I would do some manual testing with an actual instance, and then create unit tests that mock the response values in the roundtripper or RPC client. But that doesn't seem to match the current testing approach.
